source/Calamari.Tests/KubernetesFixtures/Authentication/SetupKubectlAuthenticationFixture.cs

```diff
@@ -133,54 +133,7 @@ public void Execute_WithNoClusterUrl_Fails()
 
             log.Received().Error("Kubernetes cluster URL is missing");
         }
-
-        [TestCase(true)]
-        [TestCase(false)]
-        public void Execute_WithUsernamePasswordAndCertificates_ConfiguresAuthenticationCorrectly(bool skipServerCertificate)
```

Comment:
This test is actually not particularly meaningful since you can either authenticate with UsernamePassword, _or_ ClientCertificate. They are both different auth mechanisms and although im willing to assume that kubectl might not have a problem with setting multiple auth settings, as far as our data model is concerned, its one or the other.

Its this kind of mixing of mutually exclusive authentication mechanisms that have made the auth code so hard to follow and complex. 

Both support setting the _Server cert_, since that is used for TLS verification, but client cert is the authentication mechanism that the server uses to validate the client is who it says it is (in contrast to Username and passwrd)
![image](https://github.com/OctopusDeploy/Calamari/assets/1830666/b16d732c-354b-4671-840e-84c146de6bae)
![image](https://github.com/OctopusDeploy/Calamari/assets/1830666/d2917c2f-061c-4b28-9981-63f8a7a74fc4)